spec/functional/resource/insserv_spec.rb

```diff
@@ -0,0 +1,177 @@
+# encoding: UTF-8
+#
+# Author:: Dheeraj Dubey (<dheeraj.dubey@msystechnologies.com>)
+# Copyright:: Copyright 2009-2019, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+require "functional/resource/base"
+require "chef/mixin/shell_out"
+require "fileutils"
+
+describe Chef::Resource::Service, :requires_root, :sles11 do
```

Comment:
we also probably need this functional test to run on rhel as well to catch any issues with the other case.

Reply:
Hi @lamont-granquist I've tried to verify from different OSs if there is `/sbin/insserv` but couldn't find this provider on RHEL6 and CentOs6 as you can see in my above comment https://github.com/chef/chef/pull/8256#discussion_r275746842.

So restricted the tests to SLES11 only.